wallet/wallet-cli/src/repl/mod.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::{Command, FromArgMatches, Subcommand};
+use node_comm::node_traits::NodeInterface;
+use reedline::{
+    ColumnarMenu, DefaultCompleter, DefaultHinter, DefaultValidator, EditMode, Emacs,
+    ExampleHighlighter, FileBackedHistory, ListMenu, Reedline, ReedlineMenu, Signal, Vi,
+};
+
+use crate::{
+    cli_println,
+    commands::{handle_wallet_command, WalletCommands},
+    config::WalletCliConfig,
+    errors::WalletCliError,
+    output::OutputContext,
+    repl::wallet_prompt::WalletPrompt,
+    DefWallet,
+};
+
+mod wallet_prompt;
+
+const HISTORY_FILE_NAME: &str = "history.txt";
+const HISTORY_MAX_LINES: usize = 1000;
+
+fn get_repl_command() -> Command {
+    // Strip out usage
+    const PARSER_TEMPLATE: &str = "\
+        {all-args}
+    ";
+
+    // Strip out name/version
+    const APPLET_TEMPLATE: &str = "\
+        {about-with-newline}\n\
+        {usage-heading}\n    {usage}\n\
+        \n\
+        {all-args}{after-help}\
+    ";
+
+    let repl_command = Command::new("repl")
+        .multicall(true)
+        .arg_required_else_help(true)
+        .subcommand_required(true)
+        .subcommand_value_name("APPLET")
+        .subcommand_help_heading("APPLETS")
```

Comment:
How about Commands?

Reply:
Sorry, what about it? I think we have everything we need.

Reply:
I mean the name "applets" is weird